File: languages/csharp-dotnet-core2/template/src/Algorithm.cs
using System;
using Algorithmia;

//This is an example of an algorithm. You can test it by copy/pasting the following into the web editor:
/// {"name": "Algorithmia user"}
///
/// You should expect to see the following as output:
/// {"output":"Hello Algorithmia user"}



/// 'Algo' is the primary namespace of your algorithm.
namespace Algo
{
    /// The input object type for your algorithm, the types defined here must be json serializable via the 'Newtonsoft.Json' package.
    /// Note: json serialization is 'Case Sensitive', this means that for certain types of input - it might make sense to break
    /// from C# convention, and name your AlgoInput/AlgoOutput variables in 'snake_case' format. (https://en.wikipedia.org/wiki/Snake_case)
    public class AlgoInput
    {
        
        public string name;
    }

    /// This is the output object type for your algorithm, the types defined here must be json serializable via the 'Newtonsoft.Json' package.
    public class AlgoOutput
    {
        public string greeting;
    }
    
    /// This is the main class of your algorithm, it contains your static 'apply' method which we use as an entry point to your project.
    class Algorithm
    {
        /// API calls will begin at the apply() method, with the request body passed as 'input'
        /// For more details, see algorithmia.com/developers/algorithm-development/languages
        public static AlgoOutput apply(AlgoInput input)
        {
            string name = input.name;
            AlgoOutput output = new AlgoOutput {greeting = "Hello " + name}; 
            return output;
        }
    }
}

File: python/template/src/Algorithm.py
import AlgorithmiaDevtools
import Algorithmia

# API calls will begin at the apply() method, with the request body passed as 'input'
# For more details, see algorithmia.com/developers/algorithm-development/languages


def apply(input, context=None):
    local_path = context['local_path']
    if 'metrics' in context:
        context['metrics'] += 1
    else:
        context['metrics'] = 1
    return "hello {}, the file {} is installed.".format(input, local_path)


# This is a user defined functor that mutates a 'context' object.
def download_model(context):
    url = "data://zeryx/collection/stream_file_8.mp4"
    client = Algorithmia.client()
    local_path = client.file(url).getFile().name
    print("file downloaded")
    context['local_path'] = local_path
    return context


# Hey this is important don't forget it and don't erase it
def configure():
    algorithm = AlgorithmiaDevtools.AlgorithmHandler()
    algorithm.set_apply_function(apply)
    algorithm.set_on_load_function(download_model)
    return algorithm

